src/core/locator/qgslocator.h

```diff
@@ -93,12 +99,34 @@ class CORE_EXPORT QgsLocator : public QObject
      */
     QList< QgsLocatorFilter *> filters();
 
+#ifndef SIP_RUN
+
     /**
      * Returns a map of prefix to filter, for all registered filters
      * with valid prefixes.
      * \see filters()
      */
-    QMap< QString, QgsLocatorFilter *> prefixedFilters() const;
+    QMultiMap<QString, QgsLocatorFilter *> prefixedFilters() const;
+#else
+
+    /**
+     * Returns a map of prefix to filter, for all registered filters
+     * with valid prefixes.
+     * \deprecated since QGIS 3.2 use prefixes() and filters( const QString &prefix )
+     * \see filters()
+     */
+    Q_DECL_DEPRECATED QMap<QString, QgsLocatorFilter *> prefixedFilters() const;
+#endif
+
+    /**
+      * Sets the custom prefix for a filter
+      */
+    void setCustomPrefix( QgsLocatorFilter *filter, const QString &customPrefix );
```

Comment:
Can we hide this from the bindings? I'm concerned it could be used by plugins to get around the 3 char minimum prefix requirement.